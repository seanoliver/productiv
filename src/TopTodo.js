import React from 'react';

import Todo from './Todo';

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  if (todos.length === 0) return null; // so if no todos - will not show on page

	// lowest-priority # is the highest priority
	let top = todos.reduce(
		(acc, cur) => (cur.priority < acc.priority ? cur : acc),
		todos[0]
	);

	return (
		<Todo
			title={top.title}
			description={top.description}
			priority={top.priority}
			id={top.id}
		/>
	);
}

export default TopTodo;
