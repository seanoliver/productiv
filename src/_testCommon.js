const TEST_TODOS = [{
	id: 'im-an-id',
	title: 'Test title',
	description: 'Test description',
	priority: 1,
},
{
	id: 'im-also-an-id',
	title: 'Test title 2',
	description: 'Test description 2',
	priority: 1,
},
{
	id: 'me-too',
	title: 'Test title 3',
	description: 'Test description 3',
	priority: 1,
}];

export { TEST_TODOS }