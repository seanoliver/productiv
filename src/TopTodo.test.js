import { render, fireEvent } from "@testing-library/react";
import TopTodo from "./TopTodo.js";
import { TEST_TODOS } from './_testCommon.js';


describe("TopTodo tests", function () {
  it("renders without crashing", function () {
   render(<TopTodo todos={TEST_TODOS} />);
  });

  it("matches snapshot with todos", function () {
    const { container, debug } = render(<TopTodo todos={TEST_TODOS} />);

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot no todos", function () {
    const { container, debug } = render(<TopTodo todos={[]} />);

    expect(container).toMatchSnapshot();
  });

  it("has the correct title", function () {
    const { container, debug } = render(<TopTodo todos={TEST_TODOS} />);

    const title = container.querySelector(".Todo-title");

    expect(title.textContent).toMatch("Test title");
  });

  it("has the correct description", function () {
    const { container, debug } = render(<TopTodo todos={TEST_TODOS} />);

    const description = container.querySelector(".Todo-description");

    expect(description.textContent).toMatch("Test description");
  });

  it("has the correct priority", function () {
    const { container, debug } = render(<TopTodo todos={TEST_TODOS} />);


    const priority = container.querySelector(".Todo-priority");

    expect(priority.textContent).toMatch("1");
  });

  // test no todos doesn't show top todos
  it("No TopTodo when no todos", function () {
    const { container, debug } = render(<TopTodo todos={[]} />);

    expect(container.querySelector(".TopTodo")).not.toBeInTheDocument();
  });

});