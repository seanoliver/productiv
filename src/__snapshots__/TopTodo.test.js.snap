// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo tests matches snapshot no todos 1`] = `<div />`;

exports[`TopTodo tests matches snapshot with todos 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b
        class="Todo-title"
      >
        Test title
      </b>
      <small
        class="Todo-priority"
      >
        (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small
        class="Todo-description"
      >
        Test description
      </small>
    </div>
  </div>
</div>
`;
